packages/ui/src/domainModules/collectionMammals/components/MammalManager/index.js

```diff
@@ -131,31 +131,43 @@ const mapStateToProps = (
   )
 
   const mainColumnActiveTab = getMainColumnActiveTab(url)
-  const currentTableRowNumber = keyObjectGlobalSelectors.get.currentTableRowNumber(
-    state
-  )
+
   const totalNumberOfRecords =
     specimenSearchState &&
     specimenSearchState.items &&
     specimenSearchState.items.length
 
+  const currentTableRowNumber = keyObjectGlobalSelectors.get.currentTableRowNumber(
+    state
+  )
+
   const isEditRecordView = mainColumnActiveTab === 'recordEdit'
   const isNewRecordView = mainColumnActiveTab === 'recordNew'
   const isItemViewOrSettings = mainColumnActiveTab.startsWith('record')
   const isTableView = mainColumnActiveTab === 'resultTable'
   const isTableViewOrSettings = mainColumnActiveTab.startsWith('resultTable')
 
   const showSelectNextRecordButton =
-    !isNewRecordView && currentTableRowNumber !== totalNumberOfRecords
+    !isNewRecordView && currentTableRowNumber < totalNumberOfRecords
   const showSelectPreviousRecordButton =
-    !isNewRecordView && currentTableRowNumber !== 1
+    !isNewRecordView && currentTableRowNumber > 1
```

Comment:
have you made sure this works always? because what happens if currentTableRowNumber is `undefined`?